doc/feature/restAPI.md

```diff
@@ -0,0 +1,94 @@
+
+## API
+
+Invoking a GET request on the / url returns the list of top-level namespaces, 
+e.g. 'zerobuf'. Similarily, a GET /zerobuf/render/ would return the objects in
+the zerobuf::render namespace. Finally, a GET /namespace/object/schema would 
+return the schema for the object itself.
+
+   Request: GET http://hostname:port/
+   Response: [ "namespace1", "namespace2", "..." ]
+
+   Request: GET http://hostname:port/namespace1/
+   Response: [ "object1", "object2", "..." ]
+
+   Request: GET http://hostname:port/namespace1/object1/schema
+   Response: { "object1" : { "attribute1": "string", "attribute2": "boolean" } }
+
+## Examples
+- Create an HTTP server and register serializables
+
+    void setupHTTPServer( const int argc, char** argv )
+    {
+        // Create HTTP server
+        _httpServer = zeq::http::Server::parse( argc, argv );
+        if( !_httpServer )
+            return;
+
+        // Register C++ objects
+        subscribers.push_back( _httpServer );
+        _httpServer.subscribe( camera, LUT, vrParams );
+        _httpServer.register( camera, LUT, vrParams, frame );
+    }
+
+- Query API
+   Request: GET http://hostname:port/zerobuf/render/camera/schema
+   Response: { "camera" : { "origin": "string", "lookat": "up" } }
+
+## Implementation
+
+* Add a getSchema public method to Servus::serializable
+* Make sure getSchema is added to code generator
+* Add the '/schema' url to every registered object, and make it return the
+  schema of the object in a JSON form.
+
+## Issues
+
+### API
+
+Several options are beging discussed regarding the implementation of the API:
+
+#### Option 1
+GET / returns the list of top-level namespaces, e.g., 'zerobuf', GET 
+/zerobuf/render/ returns the objects in zerobuf::render. GET 
+/namespace/object/schema returns the schema for the object.
+- Example:
+   Request: GET http://hostname:port/
+   Response: [ "namespace1", "namespace2", "..." ]
+
+   Request: GET http://hostname:port/namespace1/
+   Response: [ "object1", "object2", "..." ]
+
+   Request: GET http://hostname:port/namespace1/object1/schema
+   Response: { "object1" : { "attribute1": "string", "attribute2": "boolean" } }
+- Pros: Clean API, and a one-to-one match between a URI and the registered 
+  object.
+- Cons: Client needs to traverse the tree or URIs to get the list of registered 
+  objects. A mechanism to distinguish between an object and a namespace is 
+  needed.
+
+#### Option 2
+GET /api-docs returns the list of URIs and schemas for registered objects. 
+- Example:
+   Request: GET http://hostname:port/api-docs
+   Response: 
+   { 
+      { "namespace/foo": { foo: { "name": string } },
+      { "namespace/bar": { bar: { "x": float, "y": float, 'z": float } 
+   }
+- Pros: Simple client and server side implementations. No need for a recursive 
+  walk in the tree of URIs. Only one invocation is required to get the list of
+  all registered objects.
+- Cons: /api-docs becomes a reserved URI.
+
```

Comment:
We should already decide which of the options to implement, right? I guess you favor Option2, and it seems more prominent in the web world. I'd favor Option1 right now because of the clean semantics, but introspection is cumbersome, agreed. I'm not sure how we vote for that.